### ISSUE TITLE
Error on list containers and images

### ISSUE THREAD

**Issue by mdauri** [microsoft/vscode-docker#3249]
Issue Type: <b>Bug</b>

Since I update to the last docker client for windows version, the extension dont lists any image or container

VS Code version: Code 1.60.1 (83bd43bc519d15e50c4272c6cf5c1479df196a4d, 2021-09-10T17:07:10.714Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 3400G with Radeon Vega Graphics     (8 x 3693)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.95GB (5.34GB free)|
|Process Argv|--crash-reporter-id f85e3423-a069-40b9-8ea5-aab8aacd446a|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (13)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-css-modules|cli|0.3.2
split-html-attributes|dan|1.0.10
vscode-eslint|dba|2.1.25
gitlens|eam|11.6.0
EditorConfig|Edi|0.16.4
vscode-pull-request-github|Git|0.30.0
vscode-random|jre|1.10.0
vscode-colorize|kam|0.11.1
vscode-docker|ms-|1.17.0
remote-containers|ms-|0.194.0
remote-wsl|ms-|0.58.2
powershell|ms-|2021.9.0
material-icon-theme|PKi|4.10.0

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspyt653:30270858
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551:30345470
pythonptprofiler:30281270
vsdfh931cf:30280410
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
pythonvsuse255:30340121
vscod805:30301674
pythonvspyt200:30340761
vscextlangct:30333562
binariesv615:30325510
vsccppwtct:30364498
pythonvssor306:30344512
bridge0708:30335490
pygetstartedc2:30360494
dockerwalkthru:30370836
bridge0723:30353136
pythonrunftest32:30365366
pythonf5test824:30361777
javagetstartedt:30364666
pythonvspyt187cf:30365362
pydsgsc2:30361791
vssid140:30363603
vssur157:30367808

```

</details>

**Comment by bwateratmsft** [comment]
@mdauri can you expand the view so the full `docker context ls ...` error message is visible?

**Comment by ejwettstein** [comment]
I may be having the same problem. Here's an expanded view the error that I'm getting.
![image](https://user-images.githubusercontent.com/646090/136080535-b045d008-85d4-4353-ada5-cfc572cb5c7d.png)

Running the command from a terminal, I see.
![image](https://user-images.githubusercontent.com/646090/136080777-375e4575-1aec-4dfe-a779-c3b4cd21ff22.png)

docker context seems to broken on my machine. Other docker commands are working fine.

**Comment by bwateratmsft** [comment]
Strange. Does it work if you do just `docker context ls`, no `--format`?

Does the folder `%userprofile%\.docker` exist?

**Comment by ejwettstein** [comment]
`docker context ls` give the same error.
`docker context inspect default` works.
Yes, `%userprofile%\.docker` exists.

I also tried resetting my docker settings in docker desktop. No change.
Updated to docker desktop 4.1.0, No change.

**Comment by ejwettstein** [comment]
This was a docker issue for me. I uninstalled it and removed the docker profile directory. Reinstalled and all good.

**Comment by bwateratmsft** [comment]
This comment here makes me think maybe Docker corrupted the config.json file and it needed to be recreated: https://forums.docker.com/t/docker-daemon-error-invalid-character-x00-looking-for-beginning-of-value/18072/6

What you described would align with that...

**Comment by ejwettstein** [comment]
Yes. Docker had corrupted the context.json file. It was filled with null characters. 

Docker had crashed last week. A reinstall seemed to fix everything, except this context problem.  Removing all of the docker config files fixed everything.

**Comment by bwateratmsft** [comment]
Yeah, that'd do it. I'll keep an eye out to see if this issue reoccurs with others. Thanks for investigating!
